zk/stages/stage_sequence_execute_transactions.go

```diff
@@ -161,15 +159,26 @@ func attemptAddTransaction(
 			log.Info("BatchL2Data limit reached. Not adding last transaction", "txHash", transaction.Hash())
 		}
 	}
-	if err != nil {
-		return nil, nil, overflowNone, err
-	}
-	anyOverflow := overflow || batchDataOverflow
-	if anyOverflow && !l1Recovery {
+
+	if batchDataOverflow && !l1Recovery {
 		log.Debug("Transaction preexecute overflow detected", "txHash", transaction.Hash(), "counters", batchCounters.CombineCollectorsNoChanges().UsedAsString())
 		return nil, nil, overflowCounters, nil
 	}
 
+	if _, ok := transaction.GetSender(); !ok {
+		currentHeight := header.Number.Uint64() - 1
+		sender, err := recoverSender(cfg.chainConfig, transaction, currentHeight)
+		if err != nil {
+			// In this case it will be discarded as Mined, and kept in discardLRU cache
```

Comment:
@hexoscott @revitteth is this OK, that it will me discarded like this, or we need some special handling?

Reply:
We must discard it at this point as the transaction is invalid.  If we attempt to process it the node will panic later on and we get some strange behaviour for that block.  I'm not sure how these transactions get through the pool but they end up with a sender of 0x0 when they land which isn't right.

Reply:
@IvanBelyakoff - we need to do this sender recovery check before the call to `attemptAddTransaction`.  Ideally it shouldn't get that far in the process if the sender can't be recovered.

Reply:
I see your point. Since this case is mostly for the counters overflow situation, I wanted to do the recovery of sender after it to save some CPU resources, since it is a rare case AFAIK when sender can't be recovered. If I move to some place before we calculate the counters, then we will do an extra recovery. for every tx that will overflow the counters

Reply:
OK, the problem is it is fiddly to undo the adding of counters for a single transaction after it has been added this way.  For me it's better to quit early in the process if the sender can't be recovered and not need to deal with unwinding counters after the fact.

Reply:
There is code in stage_sequence_execute that handles this undoing of counters, so maybe you can send some kind of signal up the stack and deal with it outside of this function?  We just need to be super careful around counters is all.

Reply:
I've moved sender recovery outside of addTransaction as you suggested. As we discussed, one downside of this approach is that an extra recovery wil be done for transaction that might not be included, but that is still a great improvement, while we dont mess with counters for now and avoid possible issues. I've created a task to revise the counters processing after we revise batch/block/tx processing.